hubblestack/extmods/modules/pulsar.py

```diff
@@ -206,22 +206,65 @@ def process(configfile='salt://hubblestack_pulsar/hubblestack_pulsar_config.yaml
 
     if config.get('verbose'):
         log.debug('Pulsar beacon config (compiled from config list):\n{0}'.format(config))
-
-    # Read in existing events
+# Read in existing events
     if notifier.check_events(1):
         notifier.read_events()
         notifier.process_events()
         queue = __context__['pulsar.queue']
         if config.get('verbose'):
             log.debug('Pulsar found {0} inotify events.'.format(len(queue)))
+
+        global COUNTER, compiled_regex_list, keeper, _append
+
+        start_time = time.time()
+
         while queue:
+
+            event = queue.popleft()
+            path = event.path
+
+            while path != '/':
+                if path in config:
+                    break
+                path = os.path.dirname(path)
+            try:
+                pathname = event.pathname
+            except NameError:
+                pathname = path
+
+            excludes1 = config[path].get('exclude', '')
+
+            if excludes1 and isinstance(excludes1, list):
+                for exclude1 in excludes1:
+                    if isinstance(exclude1, dict):
+                        if exclude1.values()[0].get('regex', False):
+                           try:
+                               keeper = keeper + 1
+                               compiled_regex_list[keeper] = re.compile(exclude1.keys()[0])
+                           except:
+                               log.warn('error!!')
+                               pass
+                        else:
+                            exclude1 = exclude1.keys()[0]
+                    elif '*' in exclude1:
+                        if fnmatch.fnmatch(event.pathname, exclude1):
+                            _append = False
+                    else:
+                        if event.pathname.startswith(exclude1):
+                            _append = False
+            break
+
+
+        while queue:
+
+            COUNTER = COUNTER + 1
```

Comment:
Another one that is good for benchmarking but I don't think is necessary for our final optimized version.